File: CheckMyStar.Bll.Models/InvoiceModel.cs
namespace CheckMyStar.Bll.Models
{
    public class InvoiceModel
    {
        public int Identifier { get; set; }
        public string InvoiceNumber { get; set; } = string.Empty;
        public int? QuoteIdentifier { get; set; }
        public int? ClientUserIdentifier { get; set; }
        public UserModel? ClientUser { get; set; }
        public int? ClientAddressIdentifier { get; set; }
        public AddressModel? ClientAddress { get; set; }
        public int? CompanySocietyIdentifier { get; set; }
        public SocietyModel? CompanySociety { get; set; }
        public int? CompanyAddressIdentifier { get; set; }
        public AddressModel? CompanyAddress { get; set; }
        public DateTime InvoiceDate { get; set; }
        public decimal TotalAmountHT { get; set; }
        public decimal TotalVATAmount { get; set; }
        public decimal TotalAmountTTC { get; set; }
        public int? PaymentStatusIdentifier { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? DueDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public List<InvoiceLineModel> InvoiceLines { get; set; } = new List<InvoiceLineModel>();
    }
}

File: CheckMyStar.Angular/src/app/www/20_Models/BackOffice/Invoice.model.ts
import { UserModel } from '../Common/User.model';
import { AddressModel } from '../Common/Address.model';
import { SocietyModel } from './Society.model';

export interface InvoiceLineModel {
    identifier: number,
    invoiceIdentifier: number,
    description: string,
    quantity: number,
    unit: string,
    unitPriceHT: number,
    vatRate: number,
    createdDate?: string,
    updatedDate?: string,
}

export interface InvoiceModel {
    identifier: number,
    invoiceNumber?: string,
    quoteIdentifier?: number,
    clientUserIdentifier?: number,
    clientUser?: UserModel | null,
    clientAddressIdentifier?: number,
    clientAddress?: AddressModel | null,
    companySocietyIdentifier?: number,
    companySociety?: SocietyModel | null,
    companyAddressIdentifier?: number,
    companyAddress?: AddressModel | null,
    invoiceDate?: string,
    totalAmountHT?: number,
    totalVATAmount?: number,
    totalAmountTTC?: number,
    paymentStatusIdentifier?: number,
    createdDate?: string,
    dueDate?: string,
    updatedDate?: string,
    invoiceLines?: InvoiceLineModel[],
    isActive?: boolean,

    // Legacy fields kept for compatibility with any remaining usages.
    number?: string,
    amount?: number,
    isPaid?: boolean,
}